code/lib/preview-api/src/modules/store/csf/portable-stories.test.ts

```diff
@@ -14,8 +21,57 @@ describe('composeStory', () => {
     },
   };
 
+  it('should call and compose loaders data', async () => {
+    const loadSpy = vi.fn();
+    const args = { story: 'story' };
+    const LoaderStory: Story = {
+      args,
+      render: (_args, { loaded }) => {
+        expect(loaded).toEqual({ foo: 'bar' });
+      },
+      loaders: [
+        async (context) => {
+          loadSpy();
+          expect(context.args).toEqual(args);
+          return {
+            foo: 'bar',
+          };
+        },
+      ],
+    };
+
+    const composedStory = composeStory(LoaderStory, {});
+    await composedStory.load();
+    expect(loadSpy).toHaveBeenCalled();
+    composedStory();
+  });
+
+  it('should work with spies set up in loaders', async () => {
+    const spyFn = vi.fn();
+
+    const Story: Story = {
+      args: {
+        spyFn,
+      },
+      loaders: [
+        async () => {
+          spyFn.mockReturnValue('mockedData');
+        },
+      ],
+      render: (args) => {
+        const data = args.spyFn();
+        expect(spyFn).toHaveBeenCalled();
+        expect(data).toBe('mockedData');
+      },
+    };
+
+    const composedStory = composeStory(Story, {});
+    await composedStory.load();
+    composedStory();
```

Comment:
On second thought, let's move the `expect(spyFn).toHaveBeenCalled();` to top level so that the test will fail if `render()` is never called for some reason.

```suggestion
      render: (args) => {
        const data = args.spyFn();
        expect(data).toBe('mockedData');
      },
    };

    const composedStory = composeStory(Story, {});
    await composedStory.load();
    composedStory();
    expect(spyFn).toHaveBeenCalled();
```